public/app/plugins/datasource/graphite/configuration/ConfigEditor.tsx

```diff
@@ -1,20 +1,99 @@
-import React from 'react';
-import { DataSourceHttpSettings } from '@grafana/ui';
-import { DataSourcePluginOptionsEditorProps } from '@grafana/data';
-import { GraphiteDetails } from './GraphiteDetails';
-import { GraphiteOptions } from '../types';
-
-export const ConfigEditor = (props: DataSourcePluginOptionsEditorProps<GraphiteOptions>) => {
-  const { options, onOptionsChange } = props;
-
-  return (
-    <>
-      <DataSourceHttpSettings
-        defaultUrl="http://localhost:8080"
-        dataSourceConfig={options}
-        onChange={onOptionsChange}
-      />
-      <GraphiteDetails value={options} onChange={onOptionsChange} />
-    </>
-  );
-};
+import React, { PureComponent } from 'react';
+import { DataSourceHttpSettings, FormLabel, Button } from '@grafana/ui';
+import { DataSourcePluginOptionsEditorProps, onUpdateDatasourceJsonDataOptionSelect } from '@grafana/data';
+import { GraphiteOptions, GraphiteType } from '../types';
+import { Select } from '@grafana/ui/src/components/Forms/Select/Select';
+
+const graphiteVersions = [
+  { label: '0.9.x', value: '0.9' },
+  { label: '1.0.x', value: '1.0' },
+  { label: '1.1.x', value: '1.1' },
+];
+
+const graphiteTypes = Object.keys(GraphiteType).map((key: string) => ({
+  label: key,
+  value: (GraphiteType as any)[key],
+}));
```

Comment:
```ts
const graphiteTypes = Object.entries(GraphiteType).map(([label: string, value: any]) => ({
  label,
  value,
}));
```

Reply:
@stevenvachon it's not in that format.

Reply:
```
export enum GraphiteType {
  Default = 'default',
  Metrictank = 'metrictank',
}
```

Reply:
That's  the same as `{ Default: 'default', Metrictank: 'metrictank' }`, though, right? `Object.entries()` will convert that to:
```js
[ ['Default', 'default'], ['Metrictank', 'metrictank'] ]
```

where we can destructure to `label` and `value` (within `Array::map`), then reformat to `{ label, value }` (via `Object.fromEntries()`).

Reply:
@stevenvachon I tried it but it didn't work. That was my theory for why.

Reply:
Any runtime or compile errors?